### ISSUE TITLE
Social Feed

### ISSUE THREAD

**Issue by ChrisPichlerEMU** [COSC481W-2024Fall/Ript-Fitness#47]
As a user of the Ript Fitness app, I want to be able to share my workout information with other users so they can like and comment on it.

As a user of the Ript Fitness app, I want to be able to view other users' workout information so I can like and comment on what they are doing.

As a user of the Ript Fitness app, I would like to make posts on a social feed to share with other users as well as view other's posts so I can like and comment on them.

Feature name - Social Feed
Description - Creates backend capability to allow for users to make posts such as workouts, text, images, and other exercise-related information to allow other users to view it, like it, and comment on it. Also allows for the capability to allow other users to view, like, and comment on other's posts on the social feed.

Task list -

Define a controller with endpoints for CRUD operations for a social feed post.
Define service classes that use the controller endpoints to incorporate the business logic to actually do the logic for the endpoints to successfully return the correct data to the front end, such as what user is making the post, how many likes/comments are on the post, and viewing the comments on the post.
Define database queries in a Repository interface to manipulate the database based on the endpoint called and the data input with the HTTP request, such as adding a post, editing a post's stats (likes/comments), and the actual content of the post.
Define a Post data model that includes number of likes, number of comments, and a user ID.
Write unit/integration tests to meet the 70%+ code coverage required to merge into the main branch.
Confirm that code has been successfully deployed on Azure once code is merged into main.
Acceptance tests -

If a user attempts to create a post on the social feed, they are able to store it into the database.
If a user wants to edit a post's content, they are able to use the UI to edit the post, change it in the database, and have that edit be the new content of the post.
If a user wants to delete a post, they can delete it and have it unviewable for other users.
If a user would like to view other users' posts, they are able to have a feed of different posts that other user's have made.
A user can successfully like and comment on other users' posts, with that information being viewable by other users.
A user can successfully view likes and comments that other users have made on their own posts.

HTTP Endpoints (may be changed later on):

Add a Post:
Type: POST
URL: localhost:8080/socialFeed/addPost
Body: PostDtoObject

Get a Post:
Type: GET
URL: localhost:8080/socialFeed/getPost/{insert PostId here}
Body: None

Edit a Post:
Type: PUT
URL: localhost:8080/socialFeed/{insert PostId here}
Body: PostDto object that will replace the current object in the database

Delete a Post:
Type: DELETE
URL: localhost:8080/socialFeed/{insert PostId here}
Body: None

Like a Post:
Type: PUT
URL: localhost:8080/socialFeed/like/{insert PostId here}/{insert UserId here}
Body: None

Comment on a Post:
Type: PUT
URL: localhost:8080/socialFeed/comment/{insert PostId here}
Body: String that represents the content of the comment

Estimated time to completion = 2 weeks
